pkg/apis/config/v1alpha2/meshrootcertificate.go

```diff
@@ -111,11 +115,69 @@ type TresorCASpec struct {
 	SecretRef corev1.SecretReference `json:"secretRef"`
 }
 
-// MeshRootCertificateStatus defines the status of the MeshRootCertificate resource
+// MeshRootCertificateIntent specifies the intent of the MeshRootCertificate
+// can be (Active, Passive).
+type MeshRootCertificateIntent string
+
+// MeshRootCertificateComponentStatus specifies the status of the certificate component,
+// can be (`True`, `False`, `Unknown`).
+type MeshRootCertificateComponentStatus string
+
+// MeshRootCertificateConditionStatus specifies the status of the MeshRootCertificate condition,
+// one of (`True`, `False`, `Unknown`).
+type MeshRootCertificateConditionStatus string
+
+// MeshRootCertificateConditionType specifies the type of the condition,
+// one of (`Ready`, `Accepted`, `IssuingRollout`, `ValidatingRollout`, `IssuingRollback`, `ValidatingRollback`).
+type MeshRootCertificateConditionType string
+
+// MeshRootCertificateComponentStatuses is the set of statuses for each certificate component in the cluster.
+type MeshRootCertificateComponentStatuses struct {
+	Webhooks        MeshRootCertificateComponentStatus `json:"webhooks"`
```

Comment:
@jaellio do you see anything incorrect or missing?

Reply:
LGTM!